### PR TITLE
raftstore: add an error-tolerant IoBytesTracker to calculate delta I/O consumption

### PR DESCRIPTION
**Pull request by hhwyt**
### What is changed and how it works?

Issue Number: Ref #15990

What's Changed:

```commit-message
This PR fixes improper error handling in calls to
get_thread_io_bytes_total and refactors related code for clarity.

get_thread_io_bytes_total can fail in two scenarios:
1. LOCAL_IO_STATS is not initialized.
2. Errors occur during ThreadId::fetch_io_bytes(), such as failing to
open files.

Previously:
• In io.rs, calls to get_thread_io_bytes_total did not handle the
  second type of error.
• In future.rs, error handling was implemented but not abstracted into
  a reusable utility.

This PR introduces IoBytesTracker, an error-tolerant utility. It starts
calculating incremental I/O bytes only after the first successful
initialization of fetch_io_bytes. Any I/O bytes consumed before
initialization are intentionally ignored.

This approach avoids larger inaccuracies by discarding potentially
unreliable data. Since io_bytes_total is a thread-local cumulative
metric, failures before the start of the statistical logic can result 
in a falsely underestimated initial value, which may lead to 
inaccurate delta calculations.
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
none
```

### CODE REVIEW

File: components/file_system/src/lib.rs [hunk 1]
```diff
@@ -54,6 +48,12 @@ use serde::{Deserialize, Deserializer, Serialize, Serializer};
 use strum::{EnumCount, EnumIter};
 use tokio::io::{AsyncRead, ReadBuf};
 
+mod file;
```
Comment (hhwyt) [inline components/file_system/src/lib.rs:51]: This is changed by formatter.

File: components/resource_control/src/future.rs [hunk 2]
```diff
@@ -142,36 +119,23 @@ impl<F: Future> Future for LimitedFuture<F> {
...
-                    IoBytes::default()
-                }
-            }
-        } else {
-            IoBytes::default()
-        };
+        let io_bytes = io_tracker
+            .as_mut()
+            .and_then(|tracker| tracker.update())
+            .unwrap_or_else(IoBytes::default);
```
Comment (glorv) [inline components/resource_control/src/future.rs:138]:
```suggestion
            .unwrap_or_default();
```

**Review by glorv — APPROVED** [review]
LGTM

**Review by LykxSassinator — APPROVED** [review]

File: components/file_system/src/lib.rs [hunk 3]
```diff
@@ -656,4 +772,46 @@ mod tests {
         reserve_space_for_recover(data_path, 0).unwrap();
         assert!(!file.exists());
     }
+
+    #[test]
+    fn test_io_bytes_tracker_normal() {
+        #[cfg(not(feature = "failpoints"))]
+        return;
```
Comment (overvenus) [inline components/file_system/src/lib.rs:779]:
```diff
     #[test]
+    #[cfg(feature = "failpoints")]
     fn test_io_bytes_tracker_normal() {
-        #[cfg(not(feature = "failpoints"))]
-        return;
```

**Pull request merged**